lib/core/src/Cardano/Wallet.hs

```diff
@@ -1388,7 +1373,13 @@ joinStakePool ctx wid (pid, pools) argGenChange pwd = db & \DBLayer{..} -> do
     walMeta <- mapExceptT atomically $ withExceptT ErrJoinStakePoolNoSuchWallet $
         withNoSuchWallet wid $ readWalletMeta (PrimaryKey wid)
 
-    when (walMeta ^. #delegation == Delegating pid) $
+    let checkAlreadyJoined (WalletDelegation (Delegating pid') Nothing) =
+            pid' == pid
+        checkAlreadyJoined (WalletDelegation _ (Just (WalletDelegationNext (Delegating pid') _))) =
+            pid' == pid
+        checkAlreadyJoined _ = False
```

Comment:
What about moving this function declaration to the where close and use a lambda case instead?

- Avoid repeating the function's name
- Keep the handler's body neat

Reply:
done